chains/evm/listener/event-handler.go

```diff
@@ -37,9 +40,14 @@ func NewDepositEventHandler(eventListener EventListener, depositHandler DepositH
 	}
 }
 
-func (eh *DepositEventHandler) HandleEvent(startBlock *big.Int, endBlock *big.Int, msgChan chan []*message.Message) error {
-	deposits, err := eh.eventListener.FetchDeposits(context.Background(), eh.bridgeAddress, startBlock, endBlock)
+func (eh *DepositEventHandler) HandleEvent(ctx context.Context, startBlock *big.Int, endBlock *big.Int, msgChan chan []*message.Message) error {
+	ctxWithSpan, span := otel.Tracer("relayer-core").Start(ctx, "relayer.core.DepositEventHandler.HandleEvent")
+	defer span.End()
+	span.SetAttributes(attribute.String("startBlock", startBlock.String()), attribute.String("endBlock", endBlock.String()))
+	logger := log.With().Str("trace_id", span.SpanContext().TraceID().String()).Logger()
```

Comment:
What does this do in terms of tracing?

Reply:
This should put all the logs related to this trace_id in to the corresponding traces

Reply:
NOTE: this is only how DD works

Reply:
IMO I would probably then just use traces and not use logs for that purpose.
Because for sure some relaying partner won't use datadog at some point

Reply:
Bcs of that I am now duplicating logs with tracing events, with next iteration I will see what to do with them, first of all I will check other attributes that might work for other tracing services to aggregate logs under traces, secondly that is the reason why I am not getting rid of logs now